### PR TITLE
Feature/3010

### LINKED ISSUE #3010: Suggestion: Tree Farm
Suggestion: Tree Farms for Lumberjack
Size would be about the same size as the farmers farm. Need to craft a "stump with axe" which is the central marker (similar to the scarecrow) When placing the effected area is highlighted. In lumberjack hut you can designate fields for certain trees.  1 for oak and birch, 1 for spruce, etc. Lumberjack will only plant and chop trees in the field area if there is one. If there is no field lumberjack works in their current state.

This would allow t use lumberjacks to clear land, chop trees, but then have them only work a certain tree farm as you get your colony established so they don't chop down trees you want to keep, or plant in areas you don't want them.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/65342346-suggestion-tree-farm?utm_campaign=plugin&utm_content=tracker%2F43175463&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F43175463&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by danbudworthmead**
Closes #3010 

# Changes proposed in this pull request:
- Added lumberjack sceptre.
- The sceptre can be used to define the area in which the lumberjack can chop trees.

Still to do:
- UI needs work.
- Change to use localisation.

Review please

### CODE REVIEW

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/ldtteam/minecolonies/pull-request/3732/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ldtteam/minecolonies)

**Comment by marchermans** [comment]
Has the UI work been completed as well as the Localisation?

**Comment by danbudworthmead** [comment]
The localisation has been added to the xml but it hasn’t been translated. I think Raycomms said he was going  to do something with the UI.

**Review by marchermans — DISMISSED** [review]

**Review by Asherslab — DISMISSED** [review]

**Comment by marchermans** [comment]
@Raycoms Is this ready?

**Comment by Raycoms** [comment]
Missing a small thing with the GUI (he was waiting for my change to be
merged before doing it) but now he can do it.

Am Fr., 26. Juli 2019 um 09:05 Uhr schrieb Marc Hermans <
notifications@github.com>:

File: src/main/java/com/minecolonies/coremod/client/gui/WindowHutLumberjack.java [hunk 1]
```diff
@@ -71,6 +108,12 @@ public WindowHutLumberjack(final BuildingLumberjack.View building)
         return IColonyManager.getInstance().getCompatibilityManager().getCopyOfSaplings().stream().filter(storage -> filterPredicate.test(storage.getItemStack())).collect(Collectors.toList());
     }
 
+    @Override
+    public void onOpened()
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowHutLumberjack.java]: seems this isnt needed :)

File: src/main/java/com/minecolonies/coremod/entity/pathfinding/PathJobFindTree.java [hunk 2]
```diff
@@ -107,6 +145,38 @@ private boolean isTree(final BlockPos pos)
     {
         if (Tree.checkTree(world, pos, treesToNotCut) && Tree.checkIfInColonyAndNotInBuilding(pos, colony))
         {
+            if (startRestriction != null && endRestriction != null) {
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/entity/pathfinding/PathJobFindTree.java]: not sure if this is actually needed with the PathingJob already having restricted coords, can probably just remove this whole check

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/pathfinding/PathJobFindTree.java]: True, I'll test on that

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/pathfinding/PathJobFindTree.java]: Yeah, he added it because there was a bug in the Path job

**Pull request merged**